linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -1590,40 +1590,39 @@ object Analyzer {
 
   private class WorkTracker(implicit val ec: ExecutionContext) {
     private val pending = new AtomicInteger(0)
+    @volatile private var _allowComplete = false
```

Comment:
Are volatile semantics enough if we don't use a `synchronized` block when setting the variable? I'm never really sure what memory barrier guarantees we get in that situation.

Shouldn't we use an `AtomicBoolean` instead?

Reply:
From the doc of [java.util.concurrent.atomic](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/package-summary.html)

> In essence, the classes in this package extend the notion of volatile values, fields, and array elements to those that also provide an atomic conditional update operation of the form
> [...]
> - get has the memory effects of reading a volatile variable.
> - set has the memory effects of writing (assigning) a volatile variable.

So if we only use `get` / `set` a volatile is equivalent to an atomic.